presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/VerificationQueryRewriterFactory.java

```diff
@@ -50,19 +51,23 @@
     private final List<Property> controlTableProperties;
     private final List<Property> testTableProperties;
 
+    private final boolean rewriteNonDeterministicColumns;
+
     @Inject
     public VerificationQueryRewriterFactory(
             SqlParser sqlParser,
             TypeManager typeManager,
             @ForControl QueryRewriteConfig controlConfig,
-            @ForTest QueryRewriteConfig testConfig)
+            @ForTest QueryRewriteConfig testConfig,
+            VerifierConfig verifierConfig)
     {
         this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
         this.typeManager = requireNonNull(typeManager, "typeManager is null");
         this.controlTablePrefix = requireNonNull(controlConfig.getTablePrefix(), "controlTablePrefix is null");
         this.testTablePrefix = requireNonNull(testConfig.getTablePrefix(), "testTablePrefix is null");
         this.controlTableProperties = constructProperties(controlConfig.getTableProperties());
         this.testTableProperties = constructProperties(testConfig.getTableProperties());
+        this.rewriteNonDeterministicColumns = verifierConfig.isRewriteNonDeterministicColumns();
```

Comment:
requireNonNull

Reply:
`rewriteNonDeterministicColumns` is not mandatory to be true. but verifierConfig cannot be null, otherwise `verifierConfig.isRewriteNonDeterministicColumns()` will throw an exception.